classes/class-db-driver-wpdb.php

```diff
@@ -170,13 +170,10 @@ public function setup_storage( $plugin ) {
 	 * Purge storage.
 	 *
 	 * @param \WP_Stream\Plugin $plugin Instance of the plugin.
-	 * @return \WP_Stream\Uninstall
 	 */
 	public function purge_storage( $plugin ) {
 		$uninstall = new Uninstall( $plugin );
-		add_action( 'wp_ajax_wp_stream_uninstall', array( $uninstall, 'uninstall' ) );
-
-		return $uninstall;
+		$uninstall->uninstall();
```

Comment:
We expect that calling `purge_storage()` will actually trigger all the purge steps so we call the `uninstall()` logic here.

We also expected that whoever is calling the `purge_storage()` method has done all the necessary input validation and permission checks. This decouples the routing knowledge from the actual business logic of purging the data.